src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1273,6 +1276,8 @@ object Desugar {
             case p => Violation.violation(s"encountered unexpected pattern: $p")
           }
 
+          case exp: PExpression if info.typ(exp) == PermissionT && !ignorePerm => permissionD(ctx)(exp)
```

Comment:
Permissions are desugared using the `permissionD` method in order to avoid adding a case split in every expression where a perm value may occur (PDiv, PConversion, PNegation, etc). However, in cases where the desugaring of a perm expression is done in the same way as for every other type (e.g. desugaring a NameOperand or a function call that yields a value of perm type), we can use the already existing desugarer instead of adding these cases to `permissionD`.
Thus, the `ignorePerm` argument skips over this case and desugarers the expression according to the following cases in the `exprD` method. Without this argument, the method would loop forever.

Reply:
I removed the ignorePerm in the latest commits